sklearn/datasets/tests/test_base.py

```diff
@@ -405,6 +405,12 @@ def test_derive_folder_and_filename_from_url():
     assert folder == "example.com"
     assert filename == "file.tar.gz"
 
+    folder, filename = _derive_folder_and_filename_from_url(
+        "https://example.com/نمونه نماینده.data"
+    )
+    assert folder == "example.com"
+    assert filename == "نمونه نماینده.data"
```

Comment:
the description of `filter_filename` says "convert spaces to dashes", but here the space is left as a space.

Reply:
Indeed, I broke that in my previous commit. Good catch. This should be fixed in 8c2120aadb0b4cd8309168443be31624140841fe.

Reply:
I pushed 8c2120aadb0b4cd8309168443be31624140841fe to my `ogrisel:fetch_file` branch but it does not show up in this PR:

https://github.com/scikit-learn/scikit-learn/compare/main...ogrisel:scikit-learn:fetch_file

Maybe github PRs are lagging?

Reply:
I pushed an empty commit and now both commits show up. Weird bug, never saw that before.